### PR TITLE
bfd-1477: Added provider to v2 for part d events

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-1477](https://jira.cms.gov/browse/BFD-1477)

**User Story or Bug Summary:**

User Story:

As a DASG Customer, I want provider mapped for PDE, so I can use that field.
AC:

eob.provider, in PDE, is mapped using serviceProviderId (the pharmacy id) in v2

---

### What Does This PR Do?

Adds provider for v2 part-d-events

### What Should Reviewers Watch For?

Verify the correct number is placed for provider for part d events.. Also this PR is validated under Carin.   
[eob-validation.pdf](https://github.com/CMSgov/beneficiary-fhir-data/files/7953610/eob-validation.pdf)

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [X] Yes
      * [ ] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — DISMISSED** [review]
LGTM

**Review by mackec1 — APPROVED** [review]
LGTM

File: apps/bfd-server/dev/api-changelog.md [hunk 1]
```diff
@@ -703,4 +703,9 @@ Future updates may add `Coding.display` values for additional fields.
 	* The "FIXME this should be mapped as a valueQuantity, not a valueCoding" issues were addressed by creating a new common method for adding quantities to an extension instead of codeable concepts for these fields. The new method is called addExtensionValueQuantity in TransformerUtils.
 	* The "FIXME this should be mapped as an extension valueIdentifier instead of as a valueCodeableConcept" issues were addressed by creating a new common method for adding identifiers to an extension instead of a codeable concept for these fields. The new method is called addExtensionValueIdentifier in TransformerUtils.
 	* The "FIXME: check if this field is non-nullable and if not remove the 'if' check" issues were addressed by comparing the fields to their definition in the rif-layout-and-fhir-mapping.xlsx file. Most fields were found to be non-nullable and so the "if" check was removed.
- 
+
+## BFD-1477 Map Provider for PDE
```
Comment (christrotteradhoc) [inline apps/bfd-server/dev/api-changelog.md:707]: This should go at the top of the file. Lets provide a small snippet of an example showing the new mapping (similar to some of the other recent ones).

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java [hunk 2]
```diff
@@ -872,6 +876,19 @@ public void shouldHavePayment() throws Exception {
     assertTrue(compare.equalsDeep(eob.getPayment()));
   }
 
+  @Test
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java:879]: I know the rest of the class is un-javadoc'ed but we are moving in that direction so please add one to this new test

**Review by christrotteradhoc — DISMISSED** [review]

**Comment by christrotteradhoc** [comment]
Just noticed that one of the security questions is marked yes but there is no link to confluence. I think this might be a shortcoming of the template but would like to get some guidance from @karlmdavis on when we should check that box for FHIR mapping changes like this one and what documentation is appropriate in a case like this.

**Review by mackec1 — APPROVED** [review]
LGTM

**Review by christrotteradhoc — DISMISSED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
LGTM

**Pull request merged**